appcues/src/main/java/com/appcues/ui/ExperienceRenderer.kt

```diff
@@ -2,91 +2,109 @@ package com.appcues.ui
 
 import com.appcues.AppcuesConfig
 import com.appcues.SessionMonitor
-import com.appcues.action.ActionProcessor
-import com.appcues.action.ExperienceAction
-import com.appcues.analytics.AnalyticsEvent
 import com.appcues.analytics.AnalyticsTracker
-import com.appcues.analytics.ExperienceLifecycleEvent.StepInteraction.InteractionType
+import com.appcues.analytics.ExperienceLifecycleTracker
+import com.appcues.analytics.track
 import com.appcues.data.AppcuesRepository
 import com.appcues.data.model.Experience
 import com.appcues.data.model.ExperiencePriority.NORMAL
 import com.appcues.data.model.ExperienceTrigger
-import com.appcues.data.model.Experiment
 import com.appcues.data.model.RenderContext
 import com.appcues.statemachine.Action.EndExperience
 import com.appcues.statemachine.Action.StartExperience
 import com.appcues.statemachine.Action.StartStep
 import com.appcues.statemachine.Error
+import com.appcues.statemachine.Error.RenderContextNotActive
 import com.appcues.statemachine.State
 import com.appcues.statemachine.State.Idling
 import com.appcues.statemachine.StateMachine
 import com.appcues.statemachine.StepReference
 import com.appcues.util.ResultOf
 import com.appcues.util.ResultOf.Failure
 import com.appcues.util.ResultOf.Success
-import com.appcues.util.appcuesFormatted
+import kotlinx.coroutines.flow.MutableSharedFlow
 import kotlinx.coroutines.flow.SharedFlow
 import org.koin.core.component.KoinScopeComponent
+import org.koin.core.component.get
 import org.koin.core.component.inject
 import org.koin.core.scope.Scope
 
-@Suppress("UNUSED_PARAMETER")
 internal class ExperienceRenderer(
     override val scope: Scope,
 ) : KoinScopeComponent {
 
     // lazy prop inject here to avoid circular dependency with AnalyticsTracker
     // AnalyticsTracker > AnalyticsQueueProcessor > ExperienceRenderer(this) > AnalyticsTracker
     private val repository by inject<AppcuesRepository>()
-    private val stateMachine by inject<StateMachine>()
     private val sessionMonitor by inject<SessionMonitor>()
     private val config by inject<AppcuesConfig>()
     private val analyticsTracker by inject<AnalyticsTracker>()
-    private val actionProcessor by inject<ActionProcessor>()
+    private val lifecycleTracker by inject<ExperienceLifecycleTracker>()
+
+    private val slots: HashMap<RenderContext, StateMachine> = hashMapOf()
```

Comment:
seems ok for now but interested in how this may evolve - I think we'll actually need the reference to the state machine owner - the view for embed case - similar to how iOS is keeping the `WeakStateMachineOwning` reference here. Eventually, the embed trait will need to be able to get the view reference for a given render context. But this can evolve later.

Reply:
not sure how thats going to work yet